shapeshift/web
# feat: disable thor features for smart contract wallets

## Description

i.e savers confirm and trade preview - lending coming up next as a direct follow-up since it's still under a feature flag

## Pull Request Type


## Issue (if applicable)



N/A

## Risk



- We may wrongly block addresses that aren't smart contract wallets

## Testing



- Connect a smart contract wallet, or use Frame to inject a read-only Gnosis safe, or any smart contract wallet
- Ensure THORChain savers deposits and withdraws are disabled with an error alert explainer
- Ensure THORChain trades are not possible with a smart contract wallet, with a button state explainer

- Test the above with a regular address and ensure you're not blocked from continuing
- Test the above with assets other than EVM assets (e.g preview an UTXO, or go to the confirm step for a savers deposit/withdraw) and ensure you're not blocked from continuing

### Engineering



- ☝🏽 

### Operations



- ☝🏽 

## Screenshots (if applicable)

- Savers deposit - regular wallet

<img width="548" alt="Screenshot 2023-11-29 at 00 52 24" src="https://github.com/shapeshift/web/assets/17035424/30813241-485a-4f9d-998d-3a605428e2d2">

- Savers deposit - SC wallet


<img width="563" alt="image" src="https://github.com/shapeshift/web/assets/17035424/fd907a28-9587-42be-b437-cc22c9f61dd7">

- Savers withdraw - regular wallet

<img width="561" alt="Screenshot 2023-11-29 at 00 52 35" src="https://github.com/shapeshift/web/assets/17035424/129d7715-0472-435e-a368-5f5805f4deef">

- Savers withdraw - SC wallet 

 
<img width="559" alt="Screenshot 2023-11-29 at 00 51 38" src="https://github.com/shapeshift/web/assets/17035424/19b845e7-b905-4fdf-ab31-39f965e37993">


- THOR Swap - regular wallet

<img width="595" alt="image" src="https://github.com/shapeshift/web/assets/17035424/a9d90708-15ed-4753-849c-914573bbc0d6">
<img width="612" alt="image" src="https://github.com/shapeshift/web/assets/17035424/e44b1f71-d8db-439a-b6bd-8a791700e056">

- THOR swap - SC wallet

<img width="586" alt="Screenshot 2023-11-29 at 01 16 35" src="https://github.com/shapeshift/web/assets/17035424/571e3091-ef3f-4a1a-8d6b-7e7a6cd7a0fe">
